allocator.go

```diff
@@ -184,3 +218,66 @@ func (a *Allocator) Allocate() (ok bool, assignments map[RangeID][]NodeID) {
 	}
 	return true, res
 }
+
+// Minimize the difference absolute difference from the previous assignment and the current assignment
+func (a *Allocator) adhereToMinimalChurnConstraint() {
+
+	if !a.opts.churnOptions.WithMinimalChurn && !a.opts.churnOptions.withMaxChurn {
+		return
+	}
```

Comment:
Pulling the if conditions outside is more idiomatic.

Reply:
I prefer calling it in the function, but since this function is called in only once in the code, Pulling the if statement out to have the same consistency as the rest of the code also makes sense.